pkg/cover/canonicalizer.go

```diff
@@ -31,12 +32,17 @@ type CanonicalizerInstance struct {
 type Convert struct {
 	conversionHash map[uint32]*canonicalizerModule
 	moduleKeys     []uint32
+	discardCount   int
```

Comment:
These two variables are local to a single `convertPCs()` call, they should not stay in this struct that is used multiple times.

Reply:
If you want to avoid code duplication, you might do something along these lines

```
type convertContext struct {
  err error
  errCount int
  errPC uint32
  convert func(pc uint32) (uint32, error)
}

func (cc *convertContext) convertPC(pc uint32) (uint32, bool) {}
func (cc *convertContext) discarded() string {}
```

```
	var retCov []uint32
	convCtx := &convertContext{convert:convert.convertPC}
	for _, pc := range cov {
		if newPC, ok := convCtx.convertPC(pc); ok {
			retCov = append(retCov, newPC)
		}
	}
	if msg := retConv.discarded(); msg != "" { 
		log.Infof(4, "discarded during PC conversion: %s", msg)
	}
```

(this is just an example and there may be many errors)

Reply:
That makes sense; thanks for the catch. Done!